ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1041,22 +1041,24 @@ extension SignalType {
 			let disposable = CompositeDisposable()
 			
 			let flush = {
-				var (leftOriginal, rightOriginal) = states.modify { states in
+				var tuple: (Value, U)?
```

Comment:
If we added a `var finished: Bool = false` here and left `isFinished` like it was before, it could be set inside the `modify`. That would simplify things a bit.